app/server.go

```diff
@@ -1790,17 +1819,24 @@ func (s *Server) doLicenseExpirationCheck() {
 		return
 	}
 
-	if !license.IsPastGracePeriod() {
-		mlog.Debug("License is not past the grace period.")
-		return
-	}
-
 	users, err := s.Store.User().GetSystemAdminProfiles()
 	if err != nil {
 		mlog.Error("Failed to get system admins for license expired message from Mattermost.")
 		return
 	}
 
+	if license.IsAroundSixtyDaysToExpiration() {
+		appErr := s.sendLicenseUpForRenewalEmail(users)
```

Comment:
As discussed offline, given that this is executes in the server runtime level and in an HA environment we can end up with multiple emails, could we store a flag in database indicating if this has been sent (or a timestamp of when it was)? We do something like this work: https://github.com/mattermost/enterprise/blob/cf4adb760d3e58eab50d71428ab80da805296fd6/cloud/worker.go#L96

Reply:
Alright. Going to do that